main/input_default.cpp

```diff
@@ -876,6 +876,10 @@ void InputDefault::joy_axis(int p_device, int p_axis, const JoyAxis &p_value) {
 
 	Joypad &joy = joy_names[p_device];
 
+	if (p_axis < 0 || p_axis >= JOY_AXIS_MAX) {
```

Comment:
Ah, you can remove  the `if (p_axis < 0 || p_axis >= JOY_AXIS_MAX) ` and the return statement, `ERR_FAIL_INDEX()` already does all of that for you :)